src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -610,23 +955,86 @@ internal class PartialCaseDecision
             public readonly SyntaxNode Syntax;
             public readonly ImmutableArray<BoundDagDecision> Decisions;
             public readonly ImmutableArray<(BoundExpression, BoundDagTemp)> Bindings;
-            public readonly BoundExpression WhereClause;
+            public readonly BoundExpression WhenClause;
             public readonly LabelSymbol CaseLabel;
             public PartialCaseDecision(
                 int Index,
                 SyntaxNode Syntax,
                 ImmutableArray<BoundDagDecision> Decisions,
                 ImmutableArray<(BoundExpression, BoundDagTemp)> Bindings,
-                BoundExpression WhereClause,
+                BoundExpression WhenClause,
                 LabelSymbol CaseLabel)
             {
                 this.Index = Index;
                 this.Syntax = Syntax;
                 this.Decisions = Decisions;
                 this.Bindings = Bindings;
-                this.WhereClause = WhereClause;
+                this.WhenClause = WhenClause;
                 this.CaseLabel = CaseLabel;
             }
+
+            public override bool Equals(object obj)
```

Comment:
>Equals [](start = 33, length = 6)

Who calls this overload? Could `PartialCaseDecision` implement `IEquatable<PartialCaseDecision>` with this method throwing `ExceptionUtilities.Unreachable` instead?